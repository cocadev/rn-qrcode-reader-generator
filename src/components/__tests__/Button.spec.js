/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import {
  Button,
} from '../index';

describe('Button Component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Button onPress={null} radius={5} textColor="primary">hello</Button>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onPress callback when pressed', () => {
    const onPress = jest.fn();
    const wrapper = shallow(
      <Button onPress={onPress} radius={5} textColor="primary">hello</Button>,
    );

    wrapper.find('TouchableOpacity').first().simulate('press');

    expect(onPress).toHaveBeenCalled();
  });
});
