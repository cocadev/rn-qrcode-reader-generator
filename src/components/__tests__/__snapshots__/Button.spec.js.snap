// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onPress={null}
    radius={5}
    textColor="primary"
  >
    hello
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <LinearGradient
        colors={
          Array [
            "#FFB36F",
            "#FF7B3C",
          ]
        }
        end={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        start={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 50,
              "elevation": 1,
              "padding": 20,
            },
            Object {
              "borderRadius": 5,
            },
            undefined,
          ]
        }
      >
        <Text
          h2={true}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "textAlign": "center",
              },
              Object {
                "color": "#FC6821",
              },
            ]
          }
          useNativeDriver={true}
        >
          hello
        </Text>
      </LinearGradient>,
      "onPress": null,
      "style": Object {
        "shadowColor": "#FC6821",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.22,
        "shadowRadius": 2.22,
      },
      "testID": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          h2={true}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "textAlign": "center",
              },
              Object {
                "color": "#FC6821",
              },
            ]
          }
          useNativeDriver={true}
        >
          hello
        </Text>,
        "colors": Array [
          "#FFB36F",
          "#FF7B3C",
        ],
        "end": Object {
          "x": 1,
          "y": 0,
        },
        "start": Object {
          "x": 0,
          "y": 1,
        },
        "style": Array [
          Object {
            "borderRadius": 50,
            "elevation": 1,
            "padding": 20,
          },
          Object {
            "borderRadius": 5,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "hello",
          "h2": true,
          "style": Array [
            Object {
              "color": "#FFFFFF",
              "textAlign": "center",
            },
            Object {
              "color": "#FC6821",
            },
          ],
          "useNativeDriver": true,
        },
        "ref": null,
        "rendered": "hello",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <LinearGradient
          colors={
            Array [
              "#FFB36F",
              "#FF7B3C",
            ]
          }
          end={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          start={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 50,
                "elevation": 1,
                "padding": 20,
              },
              Object {
                "borderRadius": 5,
              },
              undefined,
            ]
          }
        >
          <Text
            h2={true}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "textAlign": "center",
                },
                Object {
                  "color": "#FC6821",
                },
              ]
            }
            useNativeDriver={true}
          >
            hello
          </Text>
        </LinearGradient>,
        "onPress": null,
        "style": Object {
          "shadowColor": "#FC6821",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.22,
          "shadowRadius": 2.22,
        },
        "testID": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            h2={true}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "textAlign": "center",
                },
                Object {
                  "color": "#FC6821",
                },
              ]
            }
            useNativeDriver={true}
          >
            hello
          </Text>,
          "colors": Array [
            "#FFB36F",
            "#FF7B3C",
          ],
          "end": Object {
            "x": 1,
            "y": 0,
          },
          "start": Object {
            "x": 0,
            "y": 1,
          },
          "style": Array [
            Object {
              "borderRadius": 50,
              "elevation": 1,
              "padding": 20,
            },
            Object {
              "borderRadius": 5,
            },
            undefined,
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "hello",
            "h2": true,
            "style": Array [
              Object {
                "color": "#FFFFFF",
                "textAlign": "center",
              },
              Object {
                "color": "#FC6821",
              },
            ],
            "useNativeDriver": true,
          },
          "ref": null,
          "rendered": "hello",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
